x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java

```diff
@@ -4,63 +4,20 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+
 package org.elasticsearch.license;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.elasticsearch.ElasticsearchException;
-import org.elasticsearch.Version;
-import org.elasticsearch.action.ActionListener;
-import org.elasticsearch.cluster.AckedClusterStateUpdateTask;
-import org.elasticsearch.cluster.ClusterChangedEvent;
-import org.elasticsearch.cluster.ClusterState;
-import org.elasticsearch.cluster.ClusterStateListener;
-import org.elasticsearch.cluster.ClusterStateUpdateTask;
-import org.elasticsearch.cluster.metadata.Metadata;
-import org.elasticsearch.cluster.service.ClusterService;
-import org.elasticsearch.cluster.service.MasterServiceTaskQueue;
-import org.elasticsearch.common.Priority;
-import org.elasticsearch.common.component.AbstractLifecycleComponent;
-import org.elasticsearch.common.component.Lifecycle;
-import org.elasticsearch.common.logging.LoggerMessageFormat;
-import org.elasticsearch.common.scheduler.SchedulerEngine;
 import org.elasticsearch.common.settings.Setting;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.time.DateFormatter;
-import org.elasticsearch.core.Nullable;
-import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.core.TimeValue;
-import org.elasticsearch.gateway.GatewayService;
 import org.elasticsearch.protocol.xpack.XPackInfoResponse;
-import org.elasticsearch.protocol.xpack.license.LicensesStatus;
-import org.elasticsearch.protocol.xpack.license.PutLicenseResponse;
-import org.elasticsearch.threadpool.ThreadPool;
-import org.elasticsearch.xpack.core.XPackPlugin;
-import org.elasticsearch.xpack.core.XPackSettings;
 
-import java.time.Clock;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
-import java.util.Locale;
-import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-/**
- * Service responsible for managing {@link LicensesMetadata}.
- * <p>
- * On the master node, the service handles updating the cluster state when a new license is registered.
- * It also listens on all nodes for cluster state updates, and updates {@link XPackLicenseState} when
- * the license changes are detected in the cluster state.
- */
-public class LicenseService extends AbstractLifecycleComponent implements ClusterStateListener, SchedulerEngine.Listener {
-    private static final Logger logger = LogManager.getLogger(LicenseService.class);
-
-    public static final Setting<License.LicenseType> SELF_GENERATED_LICENSE_TYPE = new Setting<>(
+public interface LicenseService {
+    Setting<License.LicenseType> SELF_GENERATED_LICENSE_TYPE = new Setting<>(
```

Comment:
What's the thinking behind keeping these on the interface?

Most of these settings & constants looks like they're only relevant to the way `ClusterStateLicenseService` works - we have no need for a `xpack.license.self_generated.type` in a `LicenseService`  that always generates an in memory license with fixed values.

Reply:
> What's the thinking behind keeping these on the interface?

For this PR it helps to keep the noise down since these are referenced in quite a few tests. Overall, I agree this doesn't really belong in the interface, but IMO also doesn't necessarily belong in the implementation class either since the concepts of the settings are not directly implemented to cluster state. Ideally the settings would only be registered with their corresponding implementation, but without some changes to how plugins are loaded that is not feasible since settings are constructed before the components. We would need the reverse order (or the like) to allow the implementation to dictate which settings are in use.  I think a LicenseSetting class to tease this away from the interface would be the best place for these and move them in the next PR.